pkg/codegen/dotnet/gen.go

```diff
@@ -935,7 +935,11 @@ func (mod *modContext) genResource(w io.Writer, r *schema.Resource) error {
 
 		printComment(w, prop.Comment, "        ")
 		fmt.Fprintf(w, "        [Output(\"%s\")]\n", wireName)
-		fmt.Fprintf(w, "        public Output<%s> %s { get; private set; } = null!;\n", propertyType, propertyName)
+		if prop.Plain {
+			fmt.Fprintf(w, "        public %s %s { get; private set; } = null!;\n", propertyType, propertyName)
```

Comment:
This does not work for arrays.
Putting plain on a prop that's an array generates code like this:
```
public ImmutableArray<Pulumi.Aws.Iam.Role> InstanceRoles { get; private set; } = null!;
```

This fails during compilation because ImmutableArray is a non-nullable value type:
```
 Cannot convert null to 'ImmutableArray<SecurityGroup>' because it is a non-nullable value type [/Users/flo/development/pulumi-eks/sdk/dotnet/Pulumi.Eks.csproj]
```

Reply:
Fixed it now, hopefully. Also, added a naive Go implementation.